pkg/raft/raft.go

```diff
@@ -1053,6 +1054,14 @@ func (r *raft) poll(
 }
 
 func (r *raft) Step(m pb.Message) error {
+	if buildutil.CrdbTestBuild {
+		defer func() {
+			if err := r.checkInvariants(); err != nil {
```

Comment:
This is a quick/hacky way to check some `raft` invariants on the entire CRDB test suite in CI.

Good to keep this and similar checks in `checkInvariant` at development time, until the code is polished and well covered with tests such that the invariant is obviously well maintained.

@nvanbenschoten @arulajmani tip for your upcoming raft work.